### PR TITLE
[Cherry-Pick]: Delete the user-role mapping info when deleting the user (#25988)

### PR DESCRIPTION
**Pull request by SimFG**
issue: https://github.com/milvus-io/milvus/issues/25977
pick: #25988

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

File: internal/proxy/impl.go [hunk 1]
```diff
@@ -4942,9 +4942,10 @@ func (node *Proxy) OperatePrivilege(ctx context.Context, req *milvuspb.OperatePr
 	}
 	curUser, err := GetCurUserFromContext(ctx)
 	if err != nil {
+		log.Warn("fail to get current user", zap.Error(err))
 		return &commonpb.Status{
 			ErrorCode: commonpb.ErrorCode_UnexpectedError,
-			Reason:    err.Error(),
+			Reason:    "fail to get current user, please make sure the authorizationEnabled setting in the milvus.yaml is true",
```
Comment (xiaofan-luan) [inline internal/proxy/impl.go:4948]: this error message looks weird.

Should it be something like:
please specify username if auth enabled?

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Pull request merged**